libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Resources/ResourceExplorer.cs

```diff
@@ -41,8 +42,10 @@ public class ResourceExplorer : IDisposable
         /// <summary>
         /// Initializes a new instance of the <see cref="ResourceExplorer"/> class.
         /// </summary>
-        public ResourceExplorer()
+        /// <param name="allowCycle">If allowCycle is set to true, throw an exception when detecting cycle.</param>
+        public ResourceExplorer(bool allowCycle = true)
```

Comment:
The PR is looking good overall. The only thing that doesn't feel great is having the `allowCycle` as a ResourceExplorer constructor parameter. I'll think of alternatives to this today, to see if some feels more correct. I do understand that whatever they set needs to be injectable through DI which is why it was added here. @tomlm fyi.

Reply:
I enlisted on this branch and tried a few things to avoid this parameter being passed through the resource explorer but I didn't find a way without significant refactors. The main reason, in my opinion, is that the resource explorer should focus on resources, and there should be a separate abstraction for type / json loading. That abstraction should receive the allowCycle and not the resource explorer. The resource explorer should just interact (back and forth) with the typeloading abstraction. So this is in a way a problem with this class having too many responsibilities. 

Anyways, I'm proposing some potential refactors this week for longer term with the SDK team that is taking over this code, but for now we can go with this.

Reply:
We discussed offline that as a temporary measure we'll have a ResourceExplorerOptions to avoid having so specific parameters and allow extensibility until a larger refactor takes place.